lib/js/jsbigints.nim

```diff
@@ -0,0 +1,137 @@
+
+func toLocaleString*(this: JsBigInt; locales: openArray[cstring]): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func toString*(this: JsBigInt): cstring {.importjs: "#.toString()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string = $toString(this)
+  ## Return a string representation of `JsBigInt`.
+
+func asIntN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
+
+func asUintN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+
+func `+`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".}
+   ## Same as `div` but for `JsBigInt`.
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".}
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+
+func `+=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `-=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `*=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `div=`*(x, y: JsBigInt) {.importjs: "(# /= #)".}
+
+func `mod=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `inc`*(x: JsBigInt) {.importjs: "(++#)".}
+  ## Same as `inc` but for `JsBigInt` (uses JavaScript `++BigInt()`).
+
+func `dec`*(x: JsBigInt) {.importjs: "(--#)".}
+  ## Same as `dec` but for `JsBigInt` (uses JavaScript `--BigInt()`).
+
+func `<`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".}
+
+func `<=`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".}
+
+func `==`*(x, y: JsBigInt): bool {.importjs: "(# === #)".}
+
+func `**`*(x, y: JsBigInt): JsBigInt {.importjs: "((#) $1 #)".}
+
+func `and`*(x, y: JsBigInt): JsBigInt {.importjs: "(# && #)".}
+
+func `or`*(x, y: JsBigInt): JsBigInt {.importjs: "(# || #)".}
+
+func `not`*(x: JsBigInt): JsBigInt {.importjs: "(!#)".}
+
+func `in`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(a: JsBigInt): JsBigInt {.importjs: "($1#)".}
+
+func `xor`*(x, y: JsBigInt): JsBigInt {.importjs: "(# ^ #)".}
+
+func `shl`*(a, b: JsBigInt): JsBigInt {.importjs: "(# << #)".}
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".}
+
+func inc*(a, b: JsBigInt) {.importjs: "(# += #)".}
+
+func dec*(a, b: JsBigInt) {.importjs: "(# -= #)".}
+
+func `+`*(a: JsBigInt): JsBigInt {.error.} # Can not be used by design.
```

Comment:
ok but just comment it out:
```nim
func `+`*(a: JsBigInt): JsBigInt {.error.} # Can not be used by design.
  ## https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#dont-break-asmjs
```
once you comment it out you'll also get an error anyways:
```nim
when true:
  import std/jsbigints
  let a = big"11"
  let b = +a # Error: type mismatch: got <JsBigInt>
```

Reply:
Araq does it like that, I guess is better a compile-time error than people creating their own crashing on prod and reporting wrong bugs. :shrug: 
https://github.com/nim-lang/Nim/blob/devel/lib/std/isolation.nim#L18